compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -513,7 +513,9 @@ public void visit(BLangSimpleVarRef varRefExpr) {
                         ((BLangFunction) env.enclInvokable).closureVarSymbols.add((BVarSymbol) closureVarSymbol);
                     }
                 }
-                if (env.node.getKind() == NodeKind.ARROW_EXPR) {
+                if (env.node.getKind() == NodeKind.ARROW_EXPR && !(symbol.owner instanceof BPackageSymbol)) {
+                    symbol.owner = Symbols.createInvokableSymbol(SymTag.FUNCTION, 0, null,
```

Comment:
Why does this new symbol creation needed? What is purpose for the new binary expr check? Shall we add comment on it.

Reply:
Added a comment explaining the change